src/components/BarChart.jsx

```diff
@@ -266,21 +280,17 @@ export default class BarChart extends BaseChart {
         let fullBarWidth = ((BarChart.isHorizontal(config) ?
             (height - 120) : (width - 280)) / (dataSetLength)) - 1;
 
-
         if (!isOrdinal) {
-            if (xScale === 'time' && dataSetLength === 1) {
-                fullBarWidth = 30;
-            } else if (xScale === 'time' && config.timeStep && xAxisRange[0]) {
+            if (xScale === 'time' && config.timeStep && xAxisRange[0]) {
                 fullBarWidth = this.calculateBarWidth(BarChart.isHorizontal(config), height, width,
                     (xAxisRange[1] - xAxisRange[0]), config.timeStep.toLowerCase());
             } else {
                 fullBarWidth = ((BarChart.isHorizontal(config) ?
-                    (height - 120) : (width - 280)) / ((xAxisRange[1] - xAxisRange[0] + (2 * (config.linearSeriesStep || 1))) / (config.linearSeriesStep || 1)));
+                    (height - 120) : (width - 280)) / ((dataSetLength + (2 * (config.linearSeriesStep || 1))) / (config.linearSeriesStep || 1)));
```

Comment:
This line exceeds the max line length

Reply:
Fixed with https://github.com/wso2/react-vizgrammar/pull/208